pkg/models/tenant/tenant.go

```diff
@@ -200,6 +210,50 @@ func (t *tenantOperator) ListNamespaces(user user.Info, workspace string, queryP
 	return result, nil
 }
 
+func (t *tenantOperator) Events(user user.Info, queryParam *eventsv1alpha1.Query) (*eventsv1alpha1.APIResponse, error) {
+	listEvts := authorizer.AttributesRecord{
+		User:            user,
+		Verb:            "list",
+		APIGroup:        "",
+		APIVersion:      "v1",
+		Resource:        "events",
+		ResourceRequest: true,
+	}
+	decision, _, err := t.authorizer.Authorize(listEvts)
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+
+	var workspaces []*tenantv1alpha1.Workspace
+	lw, err := t.ListWorkspaces(user, query.New())
+	if err != nil {
```

Comment:
Need to filter out the workspace that current user have the right to list events.